feat: Add documentation on configuring bucket storage in Compute (#143)
suborbital/docs: website/docs/compute/cloud-deployment/configure-storage.md

    command: ["controlplane"]

    env:
      - name: SCC_STORAGE_PATH
        value: "gs://your-storage-bucket"
      - name: GOOGLE_APPLICATION_CREDENTIALS
        value: "/etc/gcp/sa_credentials.json"

    volumeMounts:
      - name: controlplane-config
        mountPath: "/home/scn/config"
        readOnly: true
      - name: gcs-service-account-credentials-volume
        mountPath: /etc/gcp
        readOnly: true

  - name: builder
    image: suborbital/scc-builder:v0.3.0
    command: ["builder"]

    env:
      - name: SCC_STORAGE_PATH
        value: "gs://your-storage-bucket"
      - name: GOOGLE_APPLICATION_CREDENTIALS
        value: "/etc/gcp/sa_credentials.json"

    volumeMounts:
      - name: controlplane-config
        mountPath: "/home/scn/config"
        readOnly: true
      - name: gcs-service-account-credentials-volume
        mountPath: /etc/gcp
        readOnly: true

volumes:
  - name: scc-config
    configMap:
      name: scc-config
  - name: gcs-service-account-credentials-volume
    secret:
      secretName: gcs-service-account-credentials
      items:
        - key: sa_json
          path: sa_credentials.json
```

> Note the addition of `GOOGLE_APPLICATION_CREDENTIALS` to both environments of the builder and control plane containers, the `gcs-service-account-credentials-volume` volume mount to the `volumeMounts` sections of both containers, as well as the declaration of the volume itself in the `volumes` section.

Reviewer: ```suggestion
A few things to note:
- The addition of `GOOGLE_APPLICATION_CREDENTIALS` to both environments of the builder and control plane containers
- The `gcs-service-account-credentials-volume` volume mount to the `volumeMounts` sections of both containers
- The declaration of the volume itself in the `volumes` section
```

# Configure Storage

By default, Compute will store compiled functions and function source code on the local storage in your Kubernetes cluster. For greater scalability, Compute can be configured to store artifacts in cloud-based object storage like Amazon S3 or Google Cloud Storage.

> ## Kubernetes deployment

Reviewer 2: Remove this? 

It only show part of the spec and then the spec with the secret is described below. Either way we shouldn't split the k8s deployment section with docker compose in the middle.

Author: Hrm. I feel like it's kind of important to have this section since this is the only "officially" supported deployment configuration. The docker-compose version is really only meant for local development.

# Configure Storage

By default, Compute will store compiled functions and function source code on the local storage in your Kubernetes cluster. For greater scalability, Compute can be configured to store artifacts in cloud-based object storage like Amazon S3 or Google Cloud Storage.

To configure a storage bucket, provide the `SCC_STORAGE_PATH` environment variable to both the control plane and builder, e.g. `s3://my-bucket` for Amazon S3 or `gs://my-bucket` for Google Cloud Storage. For Kubernetes deployments, this is done in `.suborbital/scc-controlplane-deployment.yaml` under the `controlplane` and `builder` containers sections, and for local docker-compose deployments, this is done in `docker-compose.yaml` under the `scc-control-plane` and `scc-builder` services.

## Authentication

Bucket authentication varies between cloud providers.

### Amazon S3

> You will need to also supply the `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`, `AWS_SESSION_TOKEN` and `AWS_REGION` environment variables to both the control plane and the builder. See the [AWS authentication documentation](https://aws.github.io/aws-sdk-go-v2/docs/configuring-sdk/) for details. It is also possible to store the configuration as a Kubernetes secret, similar to the Google Cloud Storage configuration.

Reviewer 2: note: this link only covers sdk v2, by default gocloud(and thus scn) uses sdk v1 which has it's own credential sourcing strategy. 

The env vars work for both but I would just go ahed and include both. ( see the notion doc which explains how to enable sdk2 and has links for both)

    volumeMounts:
      - name: controlplane-config
        mountPath: "/home/scn/config"
        readOnly: true
      - name: gcs-service-account-credentials-volume
        mountPath: /etc/gcp
        readOnly: true

  - name: builder
    image: suborbital/scc-builder:v0.3.0
    command: ["builder"]

    env:
      - name: SCC_STORAGE_PATH
        value: "gs://your-storage-bucket"
      - name: GOOGLE_APPLICATION_CREDENTIALS
        value: "/etc/gcp/sa_credentials.json"

    volumeMounts:
      - name: controlplane-config
        mountPath: "/home/scn/config"
        readOnly: true
      - name: gcs-service-account-credentials-volume
        mountPath: /etc/gcp
        readOnly: true

volumes:
  - name: scc-config
    configMap:
      name: scc-config
  - name: gcs-service-account-credentials-volume
    secret:
      secretName: gcs-service-account-credentials
      items:
        - key: sa_json
          path: sa_credentials.json
```

A few things to note:

- The addition of `GOOGLE_APPLICATION_CREDENTIALS` to both environments of the builder and control plane containers
- The `gcs-service-account-credentials-volume` volume mount to the `volumeMounts` sections of both containers

> - The declaration of the volume itself in the `volumes` section

Reviewer 2: docker compose section appears to be removed. I think we'd still like it just not in between two k8s sections

Reviewer 2: spoke off line, not necessary to include
